src/components/calculator.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import './calculator.css';
+import calculate from '../logic/calculate';
+
+class Calculator extends React.Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      total: 0,
+      next: null,
+      operation: null,
+    };
+
+    this.handleEvent = this.handleEvent.bind(this);
+  }
+
+  handleEvent = (buttonName) => {
+    this.setState((state) => calculate(state, buttonName));
+  };
+
+  render() {
+    const { total, operation, next } = this.state;
+    return (
+      <div>
+        <p className="result">
+          <span>{total}</span>
+          <span>{operation}</span>
+          <span>{next}</span>
+        </p>
+        <div className="calculator">
+          <button type="button" className="grayButton" name="AC" onClick={(e) => this.handleEvent(e.target.name)}>AC</button>
+          <button type="button" className="grayButton" name="+/-" onClick={(e) => this.handleEvent(e.target.name)}>+/-</button>
+          <button type="button" className="grayButton" name="%" onClick={(e) => this.handleEvent(e.target.name)}>% </button>
```

Comment:
- [ ] when I try any number modulo zero the app crash. Kindly fix the issue maybe try displaying a message error like when the user try dividing by zero

Reply:
resolved

Reply:
Thank for the review @safafa